man/sd_id128_get_machine.xml

```diff
@@ -83,11 +89,15 @@
     for more information. This function also internally caches the
     returned ID to make this call a cheap operation.</para>
 
-    <para>Note that <function>sd_id128_get_boot()</function> always
-    returns a UUID v4 compatible ID.
-    <function>sd_id128_get_machine()</function> will also return a
-    UUID v4-compatible ID on new installations but might not on older.
-    It is possible to convert the machine ID into a UUID v4-compatible
+    <para><function>sd_id128_get_invocation()</function> returns the invocation ID of the currently executed
+    service. This reads and parses the <varname>$INVOCATION_ID</varname> environment variable that the service managers
+    sets when activating a service, see
+    <citerefentry><refentrytitle>systemd.exec</refentrytitle><manvolnum>5</manvolnum></citerefentry> for details. The
+    ID is cached internally.</para>
```

Comment:
I wonder if we should leave the door open to add different mechanisms of passing the id to the process? In the description you mention generating the id from cgroup hash… If that is implemented later on, we might want to skip setting of the env var and make sd_id128_get_invocation figure out the id on its own. So maybe this text should be extended to say that "Currently sd_id128_get_invocation queries $INVOCATION_ID, but a different mechanism might be added in the future.".